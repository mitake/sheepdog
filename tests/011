#!/bin/bash

# Test sheep booting on existing images mounted as loop device

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

trap "_cleanup_devices; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup_devices()
{
    _kill_all_collies
    _kill_all_sheeps

    for i in 0 1 2; do
        umount $STORE/$i
        rm -f $STORE/$i.img
    done
    rm $seq.full
}

_cleanup

dd if=/dev/zero of=$STORE/0.img seek=$((2 * 1024 ** 3 - 1)) bs=1 count=1 > $seq.full 2>&1
dd if=/dev/zero of=$STORE/1.img seek=$((4 * 1024 ** 3 - 1)) bs=1 count=1 > $seq.full 2>&1
dd if=/dev/zero of=$STORE/2.img seek=$((8 * 1024 ** 3 - 1)) bs=1 count=1 > $seq.full 2>&1

fstype="$(df -T / | tail -1 | awk '{print $2}')"
case $fstype in
    ext2)
        mkops="-t $fstype -q -F"
        ;;
    ext[3-4])
        mkops="-t $fstype -q -F -J size=4"
        ;;
    xfs)
        mkops="-t $fstype -f"
        ;;
    *)
        _die  "For now we only support xfs and ext*."
        ;;
esac

for i in 0 1 2; do
    mkfs $mkops $STORE/$i.img > $seq.full
    mkdir $STORE/$i
    mount -o loop $STORE/$i.img $STORE/$i
    _start_sheep $i
done

sleep 1

echo check the number of vnodes
for i in 0 1 2; do
    $COLLIE node list -p 700$i
done

status=0
