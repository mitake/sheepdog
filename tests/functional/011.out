QA output created by 011
check the number of vnodes
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	27          0
   1   127.0.0.1:7001      	55          1
   2   127.0.0.1:7002      	110          2
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	27          0
   1   127.0.0.1:7001      	55          1
   2   127.0.0.1:7002      	110          2
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	27          0
   1   127.0.0.1:7001      	55          1
   2   127.0.0.1:7002      	110          2
